api/internal/localizer/localizer.go

```diff
@@ -127,3 +134,43 @@ func (lc *Localizer) localizeFile(path string) (string, error) {
 	}
 	return locPath, nil
 }
+
+// localizeDir localizes root path and returns the localized path
+func (lc *Localizer) localizeDir(path string) (string, error) {
+	ldr, err := lc.ldr.New(path)
+	if err != nil {
+		return "", errors.Wrap(err)
+	}
+	defer func() { _ = ldr.Cleanup() }()
+
+	var locPath string
+	if repo := ldr.Repo(); repo != "" {
+		// TODO(annasong): check if able to add localize directory
```

Comment:
What does this mean? I'm having trouble understanding this one

Reply:
To add a localize directory here, we should check if there's a directory at this location in `scope` that shares the localize directory name. In such a case, we should return an error to avoid confounding directories.

Reply:
I forgot to add that we don't throw an error for an unreferenced directory that shares the localize directory name. It's only a problem when the local directory is referenced and copied by `localize` AND there's a remote reference.

Reply:
Will simplify this behavior, according to our discussion today.